django/utils/asyncio.py

```diff
@@ -42,10 +42,18 @@ def __init__(self, parent):
         self.parent = parent
 
     def __getattr__(self, item):
-        original = getattr(self.parent, item)
-        if asyncio.iscoroutinefunction(original):
-            return original
-        return sync_to_async(original)
+        if item == 'parent':
+            # TODO: design talk necessary here
+            # adding in a way to bypass to the sync parent
+            # in the async API
+            # because I don't always want to make these attributes async
+            # if they are CPU only and the underlying library is not async/await supportable
+            return self.parent
```

Comment:
This makes sense. I didn't anticipate how we would write async overrides (like `FileBasedAsyncHelper`) when writing the initial helper. So basically, doing `self.some_method` inside the helper is equivalent to `self.a.some_method` in the parent.  

I like your solution; it's clean and explicit. Though we may want to rename `parent` to something that is clearer as an API. Maybe `original`? or `declared`? 

Another option would be to do something similar to what you're doing but for `self.s` and self.a` and ensure that we are explicit when overriding the methods in whether we want the sync or async versions. Something like:

```
if item in ['s', 'a']:
    return getattr(self.parent, item)
```

The only issue with this is that we don't know under which name the sync and async helper have been assigned. Sure, we could make those configurable on `__init__`, but that's adding more complexity.

I'm happy to go with your solution for now, unless you think there is a reason to try something else. 

Any thoughts @andrewgodwin?

Reply:
While working on something else, realized that this should already be the case. `__getattr__` only gets called for attributes that aren't already defined in the class (only if an `AttributeError` is raised). This means that calling `self.parent` should already work

Reply:
I see what you mean (https://docs.python.org/3/reference/datamodel.html#object.__getattr__), though I do recall that I hacked this in at some point because I was getting a coroutine object when I didn't expect one off of a `self.parent` attribute sometime during `_write_content`, but I don't recall exactly what. I'll dig back in to see exactly why